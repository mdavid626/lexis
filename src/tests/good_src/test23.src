/' Program 23: Zakladni operace '/

Scope
dim a as integer
dim b as integer
dim c as integer

dim a_d as double
dim b_d as double
dim c_d as double
dim result as double

dim a_s as string
dim b_s as string
dim c_s as string

/' Scitani integeru '/
a = 0
print a 				;            '0
b = 2
print b 				;            '2
c = a + b
print c				;            '2
c = 10 + 20
print c				;            '30
c = c + 20
print c				;            '50
c = 20 + c
print c				;            '70
print "\n"			;

/' Odcitani integeru '/
a = 0
print a				;            '0
b = 2
print b				;            '2
c = a - b
print c				;            '-2
c = 10 - 20
print c				;            '-10
c = c - 20
print c				;            '-30
c = 20 - c
print c				;            '50
print "\n"			;

/' Nasobeni integeru '/
a = 0
print a				;            '0
b = 2
print b				;            '2
c = a * b
print c				;            '0
c = 10 * 20
print c				;            '200
c = c * 20
print c				;            '4000
c = 20*c
print c				;            '80000
print "\n"			;

/' Deleni integeru '/
a = 0
print a				;            '0
b = 2
print b				;            '2
c = a \ b
print c				;            '0
c = 30 \ 20
c = c*101
print c				;            '1
c = c \ 20
print c				;            '5
c = 20 \ c
print c				;            '4
print "\n"			;







/' konkatenace retezcu '/
print b_s				;            '
c_s = A_S + B_S
print c_s				;            '
b_s = b_s + "abcd"
print b_s				;            'abcd
b_s = "efgh" + "ijkl"
print b_s				;            'efghijkl
b_s = "ijkl" + ""
print b_s				;            'ijkl
b_s = "ijkl" + b_s		
print b_s				;            'ijklijkl
print "\n"				;


' DOUBLY
a_d = 132.0
b_d = 25.0
c_d = 42.0

result = (((a_d * b_d + 22.0 ) * (c_d/2.0 + b_d) / (c_d - b_d)))
print result;
print "\n";



end scope
