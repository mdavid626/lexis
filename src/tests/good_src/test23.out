 0 2 2 30 50 70
 0 2-2-10-30 50
 0 2 0 200 4000 80000
 0 2 0 101 5 4
abcdefghijklijklijklijkl
 8988.941176470587
